### ISSUE TITLE
<C-c> does not close FlyGrep buffer

### ISSUE THREAD

**Issue by jyscao** [wsdjeg/SpaceVim#3153]
## Environment Information

- OS: linux
- vim version: -
- neovim version: 0.3.8
- SpaceVim version: 1.3.0-dev
- SpaceVim status: 

```
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	doc/LearnVim/

nothing added to commit but untracked files present (use "git add" to track)
```

## Output of the `:SPDebugInfo!`

<details><summary> SpaceVim debug information </summary>

### SpaceVim options :

```toml
  auto_disable_touchpad = 1
  autocomplete_method = 'deoplete'
  autocomplete_parens = 1
  bootstrap_after = 'my_config#after'
  bootstrap_before = 'my_config#before'
  buffer_index_type = 4
  checkinstall = 1
  colorscheme = 'gruvbox'
  colorscheme_bg = 'dark'
  colorscheme_default = 'desert'
  commandline_prompt = '➭'
  custom_color_palette = []
  custom_plugins = [['tpope/vim-rhubarb'], ['tpope/vim-eunuch'], ['dahu/vimple'], ['AndrewRadev/linediff.vim'], ['takac/vim-hardtime'], ['vim-ctrlspace/vim-ctrlspace']]
  debug_level = 1
  default_indent = 2
  disabled_plugins = []
  enable_ale = 0
  enable_bepo_layout = 0
  enable_cursorcolumn = 0
  enable_cursorline = 1
  enable_debug = 0
  enable_googlesuggest = 0
  enable_guicolors = 1
  enable_insert_leader = 1
  enable_javacomplete2_py = 0
  enable_key_frequency = 0
  enable_language_specific_leader = 1
  enable_neocomplcache = 0
  enable_neomake = 1
  enable_os_fileformat_icon = 0
  enable_powerline_fonts = 1
  enable_statusline_bfpath = 0
  enable_statusline_mode = 1
  enable_statusline_tag = 1
  enable_tabline_filetype_icon = 1
  enable_tabline_ft_icon = 0
  enable_vimfiler_filetypeicon = 0
  enable_vimfiler_gitstatus = 0
  enable_vimfiler_welcome = 1
  enable_ycm = 0
  error_symbol = '✖'
  expand_tab = 1
  filemanager = 'defx'
  filetree_direction = 'right'
  filetype_icons = {}
  force_global_config = 1
  gitcommit_issue_icon = ''
  gitcommit_pr_icon = ''
  github_username = ''
  guifont = ''
  hiddenfileinfo = 1
  home_files_number = 6
  hosts_url = 'https://raw.githubusercontent.com/racaljk/hosts/master/hosts'
  info_symbol = 'ⓘ'
  keep_server_alive = 1
  language = ''
  leader_guide_default_group_name = ''
  leader_guide_flatten = 1
  leader_guide_hspace = 5
  leader_guide_max_size = 0
  leader_guide_position = 'botright'
  leader_guide_run_map_on_popup = 1
  leader_guide_sort_horizontal = 0
  leader_guide_submode_mappings = {'<C-C>': 'win_close'}
  leader_guide_vertical = 0
  lint_on_save = 1
  lint_on_the_fly = 0
  max_column = 120
  plugin_bundle_dir = '/home/jyscao/.cache/vimfiles/'
  plugin_manager = 'dein'
  plugin_manager_processes = 16
  plugin_name = 'incsearch-easymotion.vim'
  project_rooter_automatically = 1
  project_rooter_patterns = ['.git/', '_darcs/', '.hg/', '.bzr/', '.svn/', '.SpaceVim.d/']
  realtime_leader_guide = 1
  relativenumber = 0
  search_tools = ['rg']
  sidebar_direction = ''
  sidebar_width = 30
  smartcloseignoreft = ['tagbar', 'vimfiler', 'defx', 'SpaceVimRunner', 'SpaceVimREPL', 'SpaceVimQuickFix', 'HelpDescribe', 'VebuggerShell', 'VebuggerTerminal', 'SpaceVimTabsManager']
  smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']
  snippet_engine = 'neosnippet'
  src_root = 'E:\sources\'
  statusline_inactive_separator = 'arrow'
  statusline_iseparator = 'nil'
  statusline_left_sections = ['winnr', 'filename', 'major mode', 'search count', 'syntax checking', 'minor mode lighters']
  statusline_right_sections = ['fileformat', 'cursorpos', 'percentage']
  statusline_separator = 'arrow'
  statusline_unicode_symbols = 1
  terminal_cursor_shape = 2
  update_retry_cnt = 3
  version = '1.3.0-dev'
  vim_help_language = 'en'
  vimcompatible = 0
  warning_symbol = '⚠'
  wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'
  window_smartclose = ''
  windows_index_type = 3
  windows_leader = 's'
  windows_smartclose = 'q'
```

### SpaceVim layers :

```toml
[[layers]]
  name="autocomplete"
  auto-completion-return-key-behavior='complete'
  auto-completion-tab-key-behavior='smart'
[[layers]]
  name="checkers"
[[layers]]
  name="format"
[[layers]]
  name="edit"
[[layers]]
  name="ui"
[[layers]]
  name="core"
[[layers]]
  name="core#banner"
[[layers]]
  name="core#statusline"
[[layers]]
  name="core#tabline"
[[layers]]
  name="shell"
  default_position='top'
  default_height=30
[[layers]]
  name="lang#lua"
  foldmethod='manual'
[[layers]]
  name="lang#markdown"
[[layers]]
  name="lang#python"
[[layers]]
  name="lang#scheme"
  dialect='guile'
[[layers]]
  name="lang#vim"
[[layers]]
  name="fzf"
[[layers]]
  name="git"
  git-plugin='fugitive'
[[layers]]
  name="github"
[[layers]]
  name="sudo"
[[layers]]
  name="tools"
[[layers]]
  name="tools#zeal"
[[layers]]
  name="incsearch"
```

### SpaceVim Health checking :

SpaceVim clipboard support check report:
Checking +clipboard:
      SUCCEED!
SpaceVim environment check report:
Current progpath: nvim(/usr/bin/nvim)
version: 800
OS: linux
[shell, shellcmdflag, shellslash]: ['/bin/bash', '-c', 0]
SpaceVim lua support check report:
Checking +lua:
      Failed : Known issue, neovim do not support lua now.
SpaceVim python support check report:
Checking +python3:
      SUCCEED!
Checking +python:
      SUCCEED!

### SpaceVim runtime log :

```log
[ SpaceVim ] : logger file  does not exists, only log for current process will be shown!
[ SpaceVim ] [02:12:36] [ Info ] Startup with no argv, current dir is used: ~/.SpaceVim
[ SpaceVim ] [02:12:36] [ Info ] find local conf: /home/jyscao/.SpaceVim/.SpaceVim.d/init.vim
[ SpaceVim ] [02:12:36] [ Info ] force loadding global config >>>
[ SpaceVim ] [02:12:36] [ Info ] start to apply config [glob]
[ SpaceVim ] [02:12:36] [ Info ] bootstrap_before called
[ SpaceVim ] [02:12:38] [ Info ] Start to find root for: 
[ SpaceVim ] [02:12:38] [ Info ]         (.git/):/home/jyscao/.SpaceVim/.git/
[ SpaceVim ] [02:12:38] [ Info ]         (.SpaceVim.d/):/home/jyscao/.SpaceVim/.SpaceVim.d/
[ SpaceVim ] [02:12:38] [ Info ] change to root:/home/jyscao/.SpaceVim
[ SpaceVim ] [02:12:38] [ Info ] try to open SpaceVim welcome page
[ SpaceVim ] [02:12:38] [ Info ] change to root:/home/jyscao/.SpaceVim
[ SpaceVim ] [02:12:38] [ Info ] bootstrap_after called
[ SpaceVim ] [02:12:42] [ Info ] Start to find root for: /tmp/nvim38KgPL/3/issue_report.md
[ SpaceVim ] [02:12:42] [ Info ] change to root:/home/jyscao/.SpaceVim
```
</details>

## Screenshots

![image](https://user-images.githubusercontent.com/12605746/66773451-95d7fd00-eef1-11e9-809e-57eb99a06792.png)

**Comment by wsdjeg** [comment]
to close flygrep, you need to use ESC

**Comment by jyscao** [comment]
Yes `<Esc>` works, but on your project page for wsdjeg/FlyGrep.vim, `<C-c>`, which is more convenient to reach than `<Esc>`, is also suppose to be able to close the FlyGrep buffer.

I looked through `s:MPT._function_key` inside `~/.SpaceVim/autoload/SpaceVim/plugins/flygrep.vim` to see if it can be easily added. But I couldn't find any keys mapped to closing the buffer, which was why I asked you.

Does `<C-c>` work for you with closing the FG buffer? If so, then I have an idea for why it doesn't work for me.

I checked my neovim (v0.3.8), and it doesn't have the `nvim_open_win` API call. So when `SpaceVim#plugins#flygrep#open` gets called, it ends up in the `else` branch where a window split is created with `noautocmd rightbelow split __flygrep__`, instead of your `s:FLOATING.open_win` that extends the neovim floating window API, which I assume automatically allows `<C-c>` to close these floating windows. Or is that off the mark?
